PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl/KeyboardMovement.rbxmx

```diff
@@ -134,32 +124,62 @@ function KeyboardMovement:Enable()
 	local function onTextFocusGained(textboxFocused)
 		MasterControl:SetIsJumping(false)
 	end
-	
-	SeatJumpCn = UserInputService.InputBegan:connect(function(inputObject, isProcessed)
-		if inputObject.KeyCode == Enum.KeyCode.Backspace and not isProcessed then
-			local humanoid = getHumanoid()
-			if humanoid and (humanoid.Sit or humanoid.PlatformStand) then
-				MasterControl:DoJump()
+		
+	InputBeganCn = UserInputService.InputBegan:connect(function(inputObject, isProcessed)
+			if not isProcessed then
+				if inputObject.KeyCode == Enum.KeyCode.Backspace then
+					local humanoid = getHumanoid()
+					if humanoid and (humanoid.Sit or humanoid.PlatformStand) then
+						MasterControl:DoJump()
+					end
+				elseif inputObject.KeyCode == Enum.KeyCode.Space then
+					jumpFunc(inputObject)
+				elseif inputObject.KeyCode == Enum.KeyCode.W then
+					moveForwardFunc(inputObject)
+				elseif inputObject.KeyCode == Enum.KeyCode.S then
+					moveBackwardFunc(inputObject)
+				elseif inputObject.KeyCode == Enum.KeyCode.A then
+					moveLeftFunc(inputObject)
+				elseif inputObject.KeyCode == Enum.KeyCode.D then
+					moveRightFunc(inputObject)
+				end
+			end
+		end)
+
+	InputEndedCn = UserInputService.InputEnded:connect(function(inputObject, isProcessed)
+		if not isProcessed then
+			if inputObject.KeyCode == Enum.KeyCode.Backspace then
+				local humanoid = getHumanoid()
+				if humanoid and (humanoid.Sit or humanoid.PlatformStand) then
+					MasterControl:DoJump()
+				end
```

Comment:
This probably does not apply in the both the case of InputBegan and InputEnded. Should only be done in InputBegan.